src/frontend/src/lib/utils/qr-code.utils.ts

```diff
@@ -0,0 +1,35 @@
+import type { QrResponse, QrResult } from '$lib/types/qr-code';
+import { decodePayment } from '@dfinity/ledger-icrc';
+import { isNullish, nonNullish, type Token } from '@dfinity/utils';
+
+export const decodeQrCode = ({
+	result,
+	code,
+	requiredToken
+}: {
+	result: QrResult;
+	code?: string | undefined;
+	requiredToken?: Token | undefined;
+}): QrResponse => {
+	if (result !== 'success') {
+		return { result };
+	}
+
+	if (isNullish(code)) {
+		return { result: 'cancelled' };
+	}
+
+	const payment = decodePayment(code);
```

Comment:
can `decodePayment` throws an error? if yes, we let the consumer catch those, that's the idea?

Reply:
According to the [source code](https://github.com/dfinity/ic-js/blob/cec0dc35454c24f24957c4190332b7719d5dd65b/packages/ledger-icrc/src/utils/payment.utils.ts#L26), there is not much error to return. The unmatched code is returned as undefined and any possible error that could come from parsing the results is already treated.